### ISSUE TITLE
cuDNN BatchNorm has max bound on the batch size

### ISSUE THREAD

**Issue by soumith** [pytorch/pytorch#1004]
Fatal CUDNN_NOT_SUPPORTED error. Reported by @namdar

```
import torch
import torch.nn as nn
from torch.autograd import Variable
class TestModel(nn.Module):
    def __init__(self,inp_size):
        super(TestModel, self).__init__()
        self.encoder = nn.Sequential(
            nn.Linear(inp_size, 200),
            nn.BatchNorm1d(200),
        )
    def forward(self, x):
        x = self.encoder(x)
        return x
batch_size = 10000
inp_size = 1
x = Variable(torch.randn(batch_size,inp_size)).cuda()
m = TestModel(inp_size)
m.cuda()
out = m(x)
print(out.size())
m.eval()
out2 = m(x)
```

**Comment by colesbury** [comment]
That snippet works fine for me. It's not a contiguous issue since everything is contiguous

**Comment by soumith** [comment]
I wonder if it is because of memory bottlenecks. He runs it on a card with 6GB of memory, and maybe the error is misreported.

**Comment by apaszke** [comment]
I think @ngimel said that there's a limit on the batch size in cuDNN version we're shipping with the binaries

**Comment by soumith** [comment]
ok this should fixed in 0.1.11 then!
